checkstyle.json

```diff
@@ -517,6 +517,22 @@
 		},
 		{
 			"props": {
+				"severity": "INFO",
+				"ignorePackages": [
```

Comment:
It's convenient to put a class and its enum or typedef in the same file, especially when the extra types are just a few lines or don't get used anywhere else (like checkstyle's unit tests), so people tend to do that . 
As long as you use the "main" and additional types in combination you won't get an "unused" message. Only when you use additional types without the main type, then the check fails.

I think if a type is used separately, maybe it should have its own file. If developers want to keep everything in one file, they will have to use `ignorePackages` / `ignoreModules` or set `UnusedImport` to `IGNORE`.